cloudfoundry/azure-cloud-foundry-service/src/main/java/com/microsoft/azure/cloudfoundry/service/VcapResult.java

```diff
@@ -205,9 +206,9 @@ private void addOrReplace(MutablePropertySources propertySources,
                     .get(PROPERTY_SOURCE_NAME);
             if (source instanceof MapPropertySource) {
                 target = (MapPropertySource) source;
-                for (final String key : map.keySet()) {
-                    if (!target.containsProperty(key)) {
-                        target.getSource().put(key, map.get(key));
+                for (final Entry<String, Object> entry : map.entrySet()) {
+                    if (!target.containsProperty(entry.getKey())) {
+                        target.getSource().put(entry.getKey(), map.get(entry.getKey()));
```

Comment:
Could you elaborate the issue here?

Reply:
From performance perspective it is always better to use entryset() for iterating HashMap than keyset().

For more information on this refer [stackoverflow] (https://stackoverflow.com/a/3870093/5557885)